app/controllers/AccountManagement.scala

```diff
@@ -196,31 +197,48 @@ object ManageWeekly extends LazyLogging {
               }
               weeklyPlanInfo match {
                 case Left(error) =>
-                  logger.info(s"couldn't get new rate/currency for renewal: $error")
+                  info(s"couldn't get new rate/currency for renewal: $error")
                   Ok(views.html.account.details(None, promoCode))
                 case Right((existingCurrency, weeklyPlanInfoList)) =>
                   Ok(weeklySubscription.asRenewable.map { renewableSub =>
+                    info(s"sub is renewable - showing weeklyRenew page")
                     views.html.account.weeklyRenew(renewableSub, billingSchedule, contact, billToCountry, weeklyPlanInfoList, existingCurrency, promoCode)
                   } getOrElse {
+                    info(s"sub is not renewable - showing weeklyDetails page")
                     views.html.account.weeklyDetails(weeklySubscription, billingSchedule, contact)
                   })
               }
             }.getOrElse {
-              logger.info(s"no valid bill to country for ${weeklySubscription.id}")
+              info(s"no valid bill to country for account")
               Ok(views.html.account.details(None,promoCode))
             }
           }
         }
       }
     } else {
-      logger.info(s"don't support gifts, can't manage ${weeklySubscription.id}")
+      info(s"don't support gifts, can't manage sub")
       Future.successful(Ok(views.html.account.details(None,promoCode))) // don't support gifts (yet) as they have related contacts in salesforce of unknown structure
     }
   }
 
 }
 
-object AccountManagement extends Controller with LazyLogging with CatalogProvider {
+trait ContextLogging extends LazyLogging {
+
+  // as long as there's an implicit subscription context available, log will output the subscription name automatically
+  def info[P <: AnyPlan : Subscription](message: String) = {
+    val context = implicitly[Subscription[P]]
```

Comment:
Great idea! Let's use this as a trial and see what other info we might want to include by default. If it turns out to be helpful we could move it into membership-common(?) and make use of it more widely.

For the membership apps, including the identity-id would also be good, but I guess in the case of subs that may or may not apply...